File: app/src/main/java/name/monwf/customiuizer/subs/System_AutoBrightness.java
package name.monwf.customiuizer.subs;

import android.os.Bundle;
import android.widget.SeekBar;

import name.monwf.customiuizer.SubFragment;
import name.monwf.customiuizer.prefs.SeekBarPreference;

public class System_AutoBrightness extends SubFragment {
	SeekBarPreference minBrightness;
	SeekBarPreference maxBrightness;

	@Override
	public void onActivityCreated(Bundle savedInstanceState) {
		super.onActivityCreated(savedInstanceState);

		maxBrightness = (SeekBarPreference)findPreference("pref_key_system_autobrightness_max");
		minBrightness = (SeekBarPreference)findPreference("pref_key_system_autobrightness_min");
		minBrightness.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
			@Override
			public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
				if (!fromUser) return;
				if (maxBrightness.getValue() <= progress) maxBrightness.setValue(progress + 1);
				maxBrightness.setMinValue(progress + 1);
			}

			@Override
			public void onStartTrackingTouch(SeekBar seekBar) {}

			@Override
			public void onStopTrackingTouch(SeekBar seekBar) {}
		});
	}

}

File: app/src/main/java/tv/withaibuild/customiuizer/subs/System_AutoBrightness.kt
package tv.withaibuild.customiuizer.subs

import android.os.Bundle
import android.widget.SeekBar
import tv.withaibuild.customiuizer.SubFragment
import tv.withaibuild.customiuizer.prefs.SeekBarPreference

class System_AutoBrightness : SubFragment() {

    private var maxBrightness: SeekBarPreference? = null
    private var minBrightness: SeekBarPreference? = null

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        maxBrightness = findPreference("pref_key_system_autobrightness_max")
        minBrightness = findPreference("pref_key_system_autobrightness_min")
        minBrightness?.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                if (!fromUser) return
                if ((maxBrightness?.getValue() ?: 0) <= progress) {
                    maxBrightness?.setValue(progress + 1)
                }
                maxBrightness?.setMinValue(progress + 1)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
    }
}
